src/region/RegionAnnotations.tsx

```diff
@@ -108,22 +112,21 @@ export default class RegionAnnotations extends React.PureComponent<Props, State>
                 {/* Layer 2: Drawn (unsaved) incomplete annotation target, if any */}
                 {isCreating && (
                     <RegionCreator
-                        canDraw={canDraw}
                         className="ba-RegionAnnotations-creator"
                         onStart={this.handleStart}
                         onStop={this.handleStop}
                     />
                 )}
 
                 {/* Layer 3a: Staged (unsaved) annotation target, if any */}
-                {isCreating && staged && (
+                {isCreating && isPageMatch && staged?.shape && (
```

Comment:
is the `staged?.shape` parts of the expression still required here and L129? Or is `isPageMatch` enough?

Reply:
`isPageMatch` only checks `page` but not `shape`.